### PR TITLE
DISCO-1005: preserve pagination on collections page

### PR DESCRIPTION
**Pull request by pepopowitz**
This PR populates the browser history with the appropriate URLs during pagination of a "collection/XXX" page, and makes the Collection app respect params passed in for any filters when you first hit the page. 

It is similar to #2369, but for a different page/"app". 

While doing this work, I also noticed that there were a couple other filters that weren't being respected when passed as querystring params: `offerable` and `color`. I corrected those also.

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 30270792fdf06e4d94e1bccd5e004e7650a9a76d

https://deploy-preview-2384--artsy-reaction.netlify.com

File: src/Apps/Collect/Components/Base/CollectArtworkGrid.tsx [hunk 1]
```diff
@@ -55,7 +55,8 @@ class CollectArtworkGrid extends Component<Props, LoadingAreaState> {
         const { state } = filters
         const urlFragment = urlFragmentFromState(state, { page })
 
-        // TODO: Look into using router push w/ query params.
+        // Using window.history.pushState instead of router.push, because
+        //  we just want to add to the history, not navigate to another route.
```
Comment (damassi) [inline src/Apps/Collect/Components/Base/CollectArtworkGrid.tsx]: #so-absurdly-minor: there's an extra space before the `we`

File: src/Apps/Collect/Components/Base/CollectRefetch.tsx [hunk 2]
```diff
@@ -64,11 +64,14 @@ export class CollectRefetch extends Component<CollectRefetchProps> {
             console.error(error)
```
Comment (damassi) [inline src/Apps/Collect/Components/Base/CollectRefetch.tsx]: This should be using `Utils/logger`

File: src/Apps/Collect/Components/Base/CollectArtworkGrid.tsx (hunk 1 above)
Comment (ashfurrow) [reply src/Apps/Collect/Components/Base/CollectArtworkGrid.tsx]: That makes sense – I don't have a strong opinion on starting subsequent comment lines with spaces, personally. We could use two spaces would make it more obvious that this is done with intention – a single space kind of jumps out to me, like it's in the uncanny valley of whitespace.

Comment (pepopowitz) [reply src/Apps/Collect/Components/Base/CollectArtworkGrid.tsx]: Resolved with Ash's suggestion of adding _more_ space, to escape the uncanny valley.

**Comment by artsyit** [comment]
:rocket: PR was released in v16.8.14 :rocket:

File: src/Apps/Collect/Components/Base/CollectArtworkGrid.tsx (hunk 1 above)
Comment (pepopowitz) [reply src/Apps/Collect/Components/Base/CollectArtworkGrid.tsx]: Ha, that extra space is intentional. I prefer my multi-line comments to have that indentation after the first line, so you can clearly identify paragraphs. I think it helps readability.

I'm happy to change this if you think it's a big deal, but it was done with intention.

**Pull request merged**